### PR TITLE
ECS test suite

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY

Tests for:
* ecs_cluster
* ecs_service
* ecs_service_facts
* ecs_taskdefinition
* ecs_taskdefinition_facts

Initial test suite to allow work on ECS modules to be performed in a safer fashion

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
ECS integration tests 

##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 51475cd623) last updated 2017/12/11 13:41:43 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.14 (default, Nov  2 2017, 18:42:05) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

##### ADDITIONAL INFORMATION

ecsServiceRole and ecsInstanceRole will need to be created manually for the CI (or anyone using testing-policies) - the role runs iam_role against them to check that they meet the expected permissions, not necessarily to create them (although with the right permissions, they would be created)

### CODE REVIEW

File: test/integration/targets/ecs_cluster/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  - prepare_tests
```
Comment (gundalow) [inline test/integration/targets/ecs_cluster/meta/main.yml]: I think prepare_tests is empty

File: test/integration/targets/ecs_cluster/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+      iam_role:
+        name: ecsServiceRole
+        assume_role_policy_document: "{{ lookup('file','ecs-trust-policy.json') }}"
+        state: present
+        create_instance_profile: no
+        managed_policy:
+        - AmazonEC2ContainerServiceRole
+        <<: *aws_connection_info
+
+    - name: create an ECS cluster
```
Comment (gundalow) [inline test/integration/targets/ecs_cluster/tasks/main.yml:34]: Can we do another create after this and ensure no change (idempotent)?

File: test/integration/targets/ecs_cluster/meta/main.yml (hunk 1 above)
Comment (willthames) [reply test/integration/targets/ecs_cluster/meta/main.yml]: Lazy copy and paste on my part. setup_ec2 isn't needed either unless you then include setup_ec2/tasks/common.yml, which most targets don't do.

File: test/integration/targets/ecs_cluster/tasks/main.yml (hunk 2 above)
Comment (willthames) [reply test/integration/targets/ecs_cluster/tasks/main.yml:34]: Sure, I'll have to add FIXMEs for the things that do change (ecs_cluster is fine, ecs_service and ecs_taskdefinition are broken in this respect)

**Comment by willthames** [comment]
@gundalow ready_for_review

**Review by gundalow — APPROVED** [review]

**Review by s-hertel — APPROVED** [review]

**Review by ryansb — APPROVED** [review]

**Pull request merged**